parser/parser/include/ruby_parser/location.hh

```diff
@@ -1,26 +1,38 @@
 #ifndef RUBY_PARSER_LOCATION_HH
 #define RUBY_PARSER_LOCATION_HH
 
+#include <cassert>
 #include <cstddef>
 #include <iostream>
 #include <stdint.h>
 
+#include "token.hh"
+
 namespace ruby_parser {
 
 struct location {
-    size_t begin;
-    size_t end;
-    size_t lineStart;
+    token_t begin;
+    token_t end;
```

Comment:
I confess that I am not super-excited about making tokens reachable from more places...but I don't think we have a viable approach to making tokens *not* tied to the lifetime of the parser, so my lack of excitement doesn't really count for much.

I do like that by getting rid of `lineStart` we can make locations smaller, though!